### ISSUE TITLE
Members of descendent namespaces don't seem to inherit ancestor's scope

### ISSUE THREAD

**Issue by TYoungSL** [jamescourtney/FlatSharp#173]
Seems like there's problems with namespace scopes during compilation.
Google's FBS parser seems to handle this scenario fine.

Shouldn't the descending namespaces should be able to access the inherited namespace's symbols?

A.fbs:
```fbs
namespace Aa;
struct B { v:int; }
```

C.fbs:
```fbs
include "A.fbs";
namespace Aa.Cc;
struct D { b:B; }
```

C.fbs gives the error:
> Message=’Unable to resolve type: ‘B’. Context = ‘Aa.Cc.D’.’, Scope=$..Aa.Cc.D.b

**Comment by jamescourtney** [comment]
Types from other namespaces need to be fully qualified. So you need:

```
struct D { b : Aa.B; }
```

**Comment by Rjvs** [comment]
@jamescourtney The original code works in flatc, as does your suggestion but neither seems to work in FlatSharp.Compiler... the fully qualified version gives me the error:
> The namespace 'Aa' already contains a definition for 'B'

**Comment by jamescourtney** [comment]
Sounds good. I'll take a look at this today, time permitting with my real job!

**Comment by jamescourtney** [comment]
#174

**Comment by jamescourtney** [comment]
My understanding of the rules was off. This is a bit more complicated, and may take some time. Here's the problem:

Consider this FBS file:
```
namespace A.B; table TB {}
namespace A.B.C.B; table TB {}
namespace A.B.C; table TC { V1 : B.TB; }
```
In this file, `TC.V1` references `A.B.C.B.TB`. However, if we make this change to the FBS, we get a different answer:

```
namespace A.B; table TB {}
namespace A.B.C; table TC { V1 : B.TB; }
namespace A.B.C.B; table TB {}
```

Here, `TC.V1` references `A.B.TB`.

My assumption as to the reason for this behavior is that `flatc` reads items progressively, like the C preprocessor. So when it encounters `B.TB` and `A.B.C.B.TB` is not yet defined, the best match is the parent namespace of `B.TB`.

This isn't trivial because FlatSharp actually reads the full file before attempting to do any sort of resolution of types, sort of like the C# compiler, so the answer is order-independent.

This is particularly annoying because FlatSharp's behavior in this case seems more correct to me, but I digress. In the short term, I can get a fix out that improves resolution for your case, but will still not have the `flatc` behavior.

**Comment by jamescourtney** [comment]
After clarification from the FlatBuffers team, the resolution rule is:
- When a type is declared, if its fields can be resolved, they are resolved.
- If a type is unresolved, it is attempted again at the conclusion of the parsing process.

**Comment by jamescourtney** [comment]
Fixed in [5.4.1](https://github.com/jamescourtney/FlatSharp/releases/tag/5.4.1)
